src/ui-options.c

```diff
@@ -57,7 +57,7 @@ static bool get_pref_path(const char *what, int row, char *buf, size_t max)
 	my_strcat(ftmp, ".prf", sizeof(ftmp));
 
 	/* Get a filename */
-	ok = askfor_aux(ftmp, sizeof ftmp, NULL);
```

Comment:
I think that the msg() in get_filename_keypress() will overwrite the prompt if the player tries to modify the string.  I'd suggest having a different path similar to what I suggested where the game asks "Save to pref file %s? [y/n]" instead, to make it clearer what is going on.  This applies to the second chunk of the changes in this file too.